### ISSUE TITLE
Creating audience within Matomo in order to export them into Google Ads for remarketing

### ISSUE THREAD

**Issue by Chardonneaur** [piwik/piwik#22061]
Hi team,
It would be interesting to be able to export audiences within Matomo in order to import them within Google Ads for remarketing purposes.
It is typically what other web analytics offer such as:

- https://experienceleague.adobe.com/en/docs/audience-manager/user-guide/implementation-integration-guides/integrating-third-party/rlsa-integration
- https://docs.mixpanel.com/docs/cohort-sync/integrations/google-ads
- https://help.piwik.pro/support/audiences/export-audience-to-google-ads/

**Comment by Normandygirl** [comment]
I vote for this idea. Marketers will eat this up!

**Comment by Chardonneaur** [comment]
@Normandygirl I am wondering if there isn't an easy way to sort this out by using ChatGPT, i will give it a try this week.

**Comment by Chardonneaur** [comment]
I think I figured this out @Normandygirl https://youtu.be/B87nTB1ojl0

**Comment by chopsol** [comment]
Hello Ronan,

Thank you for your thoughts on remarketing via Google Ads. However, I have serious concerns about data protection, especially with regard to the GDPR in Europe.

If you want to transfer a user ID to Matomo, you must first obtain consent from the user. Without consent, this would be a violation of the GDPR. If you then transfer an email address as a user ID to Matomo and do not allow Matomo to automatically pseudonymise it, you may be in breach of data protection regulations even with consent, because you suddenly have a personal reference to the data records and any external service provider who wants to evaluate the data in Matomo must then sign an Auftragsverarbeitungsvertrag (AVV, order processing agreement) at least in Germany. But I believe this applies to the whole of Europe due to the GDPR guidelines.

I don't see any way of implementing remarketing in this case. But there may be another approach.

When you place an ad on Google and the user clicks on it, the Google click ID is passed on. This is already used in the premium extension Advertising Conversion Export. If you have asked for consent and the Visitor-ID has been created for this session, you could match future visits and the pages visited in Matomo to this click ID and possibly generate a remarketing data record.
However, I am not sure whether you can also support remarketing via gclid or only report conversions.

But an approach other than consent and gclid seems unfeasible to me from the perspective of ethical analytics and the rules of the GDPR.

Best Joachim

**Comment by Chardonneaur** [comment]
thanks @chopsol isn't it the use of this SHA 256 https://support.google.com/google-ads/answer/7474263?hl=en which makes the import legal if we get the consent?

**Comment by chopsol** [comment]
That's right. If you have consent, you can send the information to Google as SHA 265. However, the collection of personal data such as an email address in Matomo as an open data set still constitutes the processing of personal data. For this reason alone, every external service provider or freelancer must sign a data processing agreement, even if they do not actually want to work with this data. But then it is still there. I always strongly advise my customers against this and recommend activating the function for pseudonymising the user ID.
